woocommerce-services.php

```diff
@@ -1555,15 +1569,12 @@ function add_shipping_phone_to_order_fields( $fields ) {
 		}
 
 		function get_shipping_phone_from_order( $fields, $address_type, WC_Order $order ) {
-			$order_id = WC_Connect_Compatibility::instance()->get_order_id( $order );
-			if ( 'shipping' === $address_type ) {
-				$shipping_phone = get_post_meta( $order_id, '_shipping_phone', true );
-				if ( ! $shipping_phone ) {
-					$billing_address = $order->get_address( 'billing' );
-					$shipping_phone  = $billing_address['phone'];
-				}
-				$fields['phone'] = $shipping_phone;
+			if ( 'shipping' !== $address_type ) {
+				return $fields;
 			}
+
+			$fields['phone'] = $order->get_shipping_phone() ? $order->get_shipping_phone() : $order->get_billing_phone();
```

Comment:
Nice clean up. While we are at it, I would also rename the function to something like `get_shipping_or_billing_`